### PR TITLE
[Genomic Browser] Subproject Column label

### LINKED ISSUE #6295: [Genomic Browser] Subproject Column displayed as Integer instead of Label
**Describe the bug**
A. In the Profiles tab of the Genomic Browser, when downloading the Table as a CSV, the SubProject is displayed as an integer instead of a string label.

B. In the SNP tab of the Genomic Browser, the SubProject is displayed as an integer in the table instead of a string label

Test Plan Reference: https://github.com/aces/Loris/blob/master/modules/genomic_browser/test/TestPlan.md#data-download

**To Reproduce**
Steps to reproduce the behaviour (attach screenshots if applicable):
A.
1. Go to Profiles Tab of Genomic Browser
2. Click on Download as CSV.
3. Look at Subproject column of downloaded CSV.

B.
1. Go to SNP Tab of Genomic Browser
2. Look at Subproject column datatable.

A.
image

B.
image

### PR DESCRIPTION
**Pull request by laemtl**
In Genomic Browser > Profile and SNP screens, I added a parsing logic on the csv colums to allow special treatment of data (inspired by what is already implemented on the data table with [profileColumnFormatters](https://github.com/aces/Loris/blob/23.0-release/modules/genomic_browser/jsx/profileColumnFormatter.js)) and applied a Formatter on SNP > Subproject int he data table.

[Testing card](https://github.com/aces/Loris/projects/19#card-21296727)

* Resolves #6295

### CODE REVIEW

**Comment by christinerogers** [comment]
@xlecours or @spell00 do you have time to review this one?

**Comment by xlecours** [comment]
@laemtl Did you try to JOIN on the subproject table in the module query to send the subproject name instead of the subproject_id to the frontend?

**Comment by laemtl** [comment]
Great suggestion @xlecours , I pushed a new version!

**Comment by xlecours** [comment]
Did not test but the code looks good to me.

**Comment by ridz1208** [comment]
@spell00 when something passes testing please add the "Passed Manual Test" Label

**Comment by HenriRabalais** [comment]
There is already an implementation for resolving this issue by using the `getMappedCell()` function from the `FilterableDatatable` component, towards which we are trying to migrate all Filterable Datatable implementations. However, this makes use of the `FilterableDatatable` component, which the Genomic Browser is not currently using.

I haven't tested this yet, but if it works the code looks fine. But I'm also wondering if we should take the opportunity to transition the Genomic Browser over to the new `FilterableDatatable` component system since it already has a solution to this issue.

**Comment by laemtl** [comment]
@HenriRabalais, is getMappedCell() solving specifically the csv formatting issue?

**Comment by HenriRabalais** [comment]
@laemtl It was first designed to address that issue, yes. But it also resulted in addressing a larger underlying issue which was that `getFormattedCell()` was taking on the roll of both mapping and formatting data. This meant that the data was being filtered based on the mapped and formatted data, but only being downloaded as unmapped and unformatted. 

By disentangling those processes, the data can be a) filtered on unmapped data, b) both mapped and formatted to be displayed to the user and c) simply mapped when downloaded via csv.

**Comment by HenriRabalais** [comment]
@laemtl I just realized that what I'm suggesting wouldn't be able to be properly implemented until #5835 is merged.

I'll take a closer look at the code and test the PR so that we can get this through as is, and hopefully get it changed over in the future.

File: jsx/StaticDataTable.js [hunk 1]
```diff
@@ -432,7 +440,7 @@ class StaticDataTable extends Component {
 
         if (this.hasFilterKeyword(this.props.Headers[j], data)) {
           filterMatchCount++;
-          filteredData.push(this.props.Data[index[i].RowIdx]);
+          this.props.FilteredData.push(this.props.Data[index[i].RowIdx]);
```
Comment (HenriRabalais) [inline jsx/StaticDataTable.js:443]: Props should be read-only. Components should not be altering their own props.

Comment (laemtl) [reply jsx/StaticDataTable.js]: @HenriRabalais The filter works fine, and the query received the correct params (AND cohort.SubprojectID LIKE CONCAT('%', :v_cohortsubprojectid, '%')) without changes needed.

**Comment by laemtl** [comment]
@HenriRabalais The filter works fine, and the query received the correct params (AND cohort.SubprojectID LIKE CONCAT('%', :v_cohortsubprojectid, '%')) without changes needed.

**Comment by HenriRabalais** [comment]
@laemtl okay! sounds good! I'll test and then it should be good to go!

**Pull request merged**